frontend/components/panels/NotifCheckBox.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * This file is part of Search NEU and is licensed under AGPL3.
+ * See the license file in the root folder for details.
+ */
+
+import React from 'react';
+import { Checkbox } from 'semantic-ui-react';
+import user from '../user';
+import macros from '../macros';
+import Keys from '../../../common/Keys';
+
+// This file renders the checkboxes that control which sections a user signs up for
+// notifications.
+
+export default class NotifCheckBox extends Checkbox {
+  // creates a new Notification Check Box that has the section's
+  // hash in section, and sets checked to be the initial value of if the section
+  // is currently in the user or not
+  constructor(props) {
+    super(props);
+
+    const wantedSection = this.props.section;
+
+    this.state = {};
+    this.state.checked = user.hasSectionAlready(Keys.getSectionHash(wantedSection));
+    this.state.section = wantedSection;
+  }
+
+  // if the state is currently checked, uncheck, remove the section from the user's data
+  // do opposite.
+  // send data to backend
+  handleChange = () => {
```

Comment:
I know this is a p clever method syntax that auto-binds `this` to the method, but can you do the normal syntax and then bind in the constructor? Last time i checked sublime doesn't index this as a method and we should keep code consistency.... :/

Reply:
just emaCz